modules/programs/vim.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.vim;
+  defaultPlugins = [ "sensible" ];
+
+in
+
+{
+  options = {
+    programs.vim = {
+      enable = mkEnableOption "Vim";
+
+      lineNumbers = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "Weather to show line numbers.";
+      };
+
+      tabSize = mkOption {
+        type = types.nullOr types.int;
+        default = null; 
+        example = 4;
+        description = "Set tab size and shift width to a specified number of sapces.";
+      };
+
+      plugins = mkOption {
+        type = types.listOf types.str;
+        default = defaultPlugins;
+        example = literalExample ''
+          extraPlugins: [ "YankRing" ]
```

Comment:
The example shows a function taking an argument `extraPlugins`. Wouldn't `example = [ "YankRing" ];` be sufficient?